DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -169,30 +169,30 @@ public string Mention
         /// Gets this channel's children. This applies only to channel categories.
         /// </summary>
         [JsonIgnore]
-        public IEnumerable<DiscordChannel> Children
+        public IReadOnlyList<DiscordChannel> Children
         {
             get
             {
                 return !this.IsCategory
                     ? throw new ArgumentException("Only channel categories contain children.")
-                    : this.Guild._channels.Values.Where(e => e.ParentId == this.Id);
+                    : this.Guild._channels.Values.Where(e => e.ParentId == this.Id).ToList().AsReadOnly();
```

Comment:
This doesn't need `AsReadOnly`

Reply:
Without .AsReadOnly the list can be casted back to `List<T>` and modified

Reply:
Well, actually it's LINQ, so it doens't *really* matter but consistency:tm:

Reply:
Truth be told you could just do `.AsArray()` tbh